rocketarb.js

```diff
@@ -24,15 +24,31 @@ program.option('-r, --rpc <url>', 'RPC endpoint URL', 'http://localhost:8545')
        .option('-v, --bundle-file <file>', 'filename for saving the bundle before submission or reading a saved bundle', 'bundle.json')
        .option('-a, --amount <amt>', 'amount in ether to deposit', 16)
        .option('-c, --min-fee <com>', 'minimum minipool commission fee', .15)
-       .option('-y, --no-flash-loan', 'do not use the contract to make a flash loan for the arb: use capital in the node wallet instead')
-       .option('--yes', 'skip all confirmations')
-       .option('-b, --arb-contract <addr>', 'deployment address of the RocketDepositArbitrage contract', '0x1f7e55F2e907dDce8074b916f94F62C7e8A18571')
        .option('-s, --slippage <percentage>', 'slippage tolerance for the arb swap', 2)
-       .option('-up, --uni-pool <address>', 'Uniswap pool to use when using RocketUniArb', '0xa4e0faa58465a2d369aa21b3e42d43374c6f9613')
-       .option('-k, --no-swap-reth', 'keep the minted rETH instead of selling it (only works with --no-flash-loan)')
-       .option('-gm, --mint-gas-limit <gas>', 'gas limit for mint transaction (only relevant for --no-flash-loan)', 220000)
-       .option('-ga, --approve-gas-limit <gas>', 'gas limit for approve transaction (only relevant for --no-flash-loan)', 80000)
-       .option('-gs, --swap-gas-limit <gas>', 'gas limit for swap transaction (only relevant for --no-flash-loan)', 400000)
+       .option('--yes', 'skip all confirmations')
+       .addOption(
+          new Option('-fm, --funding-method <method>', 'the method to ue for funding the arbitrage.\n\
+  - with `flashLoan`, we take out an eth flash loan and then swap through whichever route gives the best arb.\n\
+  - with `uniswap`, we swap directly through a weth <-> rEth uniswap v3 pool, using the pool\'s flash loan functionaity.\n\
+  - with `self` we use eth in the local wallet to fund the arbitrage'
+  )
+         .choices(['flashLoan', 'uniswap', 'self'])
+         .default("uniswap")
+       )
+       
+       // options for --funding-method flashLoan'
+       .option('-b, --arb-contract <addr>', 'contract address to use when --funding-method = flashLoan', '0xE46BFe6F559041cc1323dB3503a09c49fb5d8828')
+
+       // options for --funding-method uniswap'
+       .option('-ub, --uni-arb-contract <addr>', 'contract address to use when --funding-method = uniswap', '0x6fCfE8c6e35fab88e0BecB3427e54c8c9847cdc2')
```

Comment:
Sort of. There isn't a way to have "conditional defaults", as far as I can tell. We could hardcode a conditional default into the code, but then the help command would be slightly misleading, because it would show a default for the contract address that wasn't correct in some circumstances.

One option is to not have a default at all, and entirely handle the default in code, at least that way the CLI would never display an incorrect default.

Reply:
ah so the issue is the default value, got it. your approach seems fine then.